torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -860,26 +877,23 @@ void propagateQuantizationOps(Block* block) {
         continue;
       }
     }
-    for (auto* output : n->outputs()) {
-      auto inputs = getPassThroughInputs(output);
-      if (inputs.size() > 0) {
-        // note that we don't need to recursively check for prim::If
-        // here because if all inputs of a prim::If is dequantized
-        // the dequantize will be factored out before we get to this
-        // point
-        bool is_dequantized = true;
-        for (auto* input : inputs) {
-          is_dequantized &= input->node()->kind() == Symbol::aten("dequantize");
+    if (isSingleInputGeneralValueAtenFunction(n)) {
+      for (auto* output : n->outputs()) {
```

Comment:
Do we need this for loop inside the `if`, `else` statements? Can it be moved outside?

Reply:
yeah, this refactor is preparing for next PR, where we do things differently for the else branch